src/main/java/org/openrewrite/java/migrate/guava/AbstractNoGuavaImmutableOf.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.guava;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public abstract class AbstractNoGuavaImmutableOf extends Recipe {
+
+    private transient final MethodMatcher IMMUTABLE_MATCHER = new MethodMatcher(getGuavaType() + " of(..)");
```

Comment:
What's the background behind adding the `transient` here? Is it because of the call to `getGuavaType` when combined with `final`?

I'm [reading up a bit](https://stackoverflow.com/questions/2968876/final-transient-fields-and-serialization) and found some conflicting reports of using transient, final, and non-trivial objects together when serialization might be involved. 

If you think it'll work fine that fine by me; but wanted to ask both to learn and get a second opinion.

Reply:
Maybe it would make sense to define a constructor in `AbstractNoGuavaImmutableOf` that takes in two arguments, and reduce `NoGuavaImmutableMapOf` to just calling that constructor with those two arguments?

Reply:
pushed just now in https://github.com/openrewrite/rewrite-migrate-java/pull/258/commits/20564abd8aba709efd30827c7913abfc218e7b95 
hope you agree; otherwise open for a revert and further discussion